lib/releases.sh

```diff
@@ -1,5 +1,51 @@
 #!/bin/bash
 
+# Requires parameters url and version. An optional parameter can be given to
+# exclude some versions. 
+# Example usage:
+# get_latest_release_from_goproxy "https://proxy.golang.org/sigs.k8s.io/cluster-api/@v/list" "v1.8." "beta|rc|pre|alpha"
+function get_latest_release_from_goproxy() {
+
```

Comment:
`get_latest_release_from_goproxy ` is as concise as I know how to make it.  The headers need to be removed to allow the `sort` to work correctly. With out removal of the headers  the the output looks like this and we can't get the latest release from it.
```
curl -si https://proxy.golang.org/sigs.k8s.io/cluster-api/@v/list | sort -rV

x-xss-protection: 0
x-frame-options: SAMEORIGIN
x-content-type-options: nosniff
vary: Sec-Fetch-Site
vary: Sec-Fetch-Mode
vary: Sec-Fetch-Dest
v1.9.0-beta.1
v1.9.0-beta.0
v1.8.5
```

The original way is currently left as a fail safe and that is why the file has so much code in it. We agreed on this with @adilGhaffarDev.

Reply:
Yes, fallback is fine. The comment above I wrote has single-liner which is basically achieving everything the function is supposed to do.

Reply:
I agree with @tuminoid . I think it's looking good now. I'm fine with keeping it as a parameterized function rather than reducing it to a one-liner; it makes it more reusable.